app/controllers/registrations_controller.rb

```diff
@@ -17,10 +17,16 @@ def cancel_email_change!
   end
 
   def destroy
-    resource.discard
-    Devise.sign_out_all_scopes ? sign_out : sign_out(resource_name)
-    set_flash_message! :notice, :destroyed
-    respond_with_navigational(resource) { redirect_to after_sign_out_path_for(resource_name), status: Devise.responder.redirect_status }
+    if resource.valid_password?(user_delete_params[:current_password])
+      resource.user_account_feedbacks.create(user: resource, feedback: user_delete_params[:feedback]) if user_delete_params[:feedback]
+      resource.discard
+      Devise.sign_out_all_scopes ? sign_out : sign_out(resource_name)
+      set_flash_message! :notice, :destroyed
+      respond_with_navigational(resource) { redirect_to after_sign_out_path_for(resource_name), status: Devise.responder.redirect_status }
+    else
+      flash.now[:alert] = "Invalid password entered. Please try again."
```

Comment:
do a toast flash message. example on how to use this is in the dashboard controller.